src/ng-xform/number-field/input-number.component.ts

```diff
@@ -0,0 +1,127 @@
+import { AfterViewInit, Component, ElementRef, Inject, LOCALE_ID, Renderer2, Input, Output, EventEmitter } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import * as math from 'mathjs';
+
+
+/**
+ * Component to generate input field form numbers
+ */
+@Component({
+  selector: 'ng-xform-input-number',
+  templateUrl: './input-number.component.html',
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: InputNumberComponent,
+    multi: true
+  }],
+})
+export class InputNumberComponent implements ControlValueAccessor, AfterViewInit {
+
+  @Output() paste = new EventEmitter();
+  @Output() keypress = new EventEmitter<KeyboardEvent>();
+  @Input() formatOptions = { notation: 'auto'};
+  @Input() inputClass = '';
+  @Input() inputStyle = '';
+  @Input() inputId = '';
+  viewModel = '';
+
+  private input: HTMLInputElement;
+  private regExpIsValidNumber: RegExp;
+  private allowedKeyCodes = [ 43, 45, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 101, 13 ];
+  private millionSeparator: string;
+  private decimalSeparator: string;
+
+  _onChange = (value: any) => { };
+  _onTouched = () => { };
+
+  constructor(private elementRef: ElementRef, private renderer: Renderer2, @Inject(LOCALE_ID) private locale: string) {
+    let decimalSymbolKeyCode;
+    if (this.isPtLocale(locale)) {
```

Comment:
Would be nice to have that on a separate class, so it could be reused.

Also, is very strange to have a `isPtLocale` in this component that should be very generic.

Reply:
I found a way to remove it. I'm going to use the common angular function to get the numeric symbols. Thanks for your note.